src/main/java/org/broadinstitute/hellbender/tools/recalibration/RecalibrationReport.java

```diff
@@ -153,19 +144,19 @@ private void initializeReadGroupCovariates(final SortedSet<String> allReadGroups
     private void parseAllCovariatesTable(final GATKReportTable reportTable, final RecalibrationTables recalibrationTables) {
         for ( int i = 0; i < reportTable.getNumRows(); i++ ) {
             final Object rg = reportTable.get(i, RecalUtils.READGROUP_COLUMN_NAME);
-            tempCOVarray[0] = requestedCovariates[0].keyFromValue(rg);
+            tempCOVarray[0] = covariates.getReadGroupCovariate().keyFromValue(rg);
             final Object qual = reportTable.get(i, RecalUtils.QUALITY_SCORE_COLUMN_NAME);
-            tempCOVarray[1] = requestedCovariates[1].keyFromValue(qual);
+            tempCOVarray[1] = covariates.getQualityScoreCovariate().keyFromValue(qual);
 
             final String covName = (String)reportTable.get(i, RecalUtils.COVARIATE_NAME_COLUMN_NAME);
-            final int covIndex = optionalCovariateIndexes.get(covName);
             final Object covValue = reportTable.get(i, RecalUtils.COVARIATE_VALUE_COLUMN_NAME);
-            tempCOVarray[2] = requestedCovariates[RecalibrationTables.TableType.OPTIONAL_COVARIATE_TABLES_START.ordinal() + covIndex].keyFromValue(covValue);
+            final Covariate cov = covariates.getCovariateByParsedName(covName);
+            tempCOVarray[2] = cov.keyFromValue(covValue);
 
             final EventType event = EventType.eventFrom((String)reportTable.get(i, RecalUtils.EVENT_TYPE_COLUMN_NAME));
             tempCOVarray[3] = event.ordinal();
```

Comment:
it's a question of boxing/unboxing.
if we had a list, it would need to be List<Integers> but NestedIntArray stores ints and covariates return ints so we'd need to box and then unbox. Seems like it's bad either way - there are no super clear benefits from removing this hardcoding. It's very local. Let me know what you think.